cumulusci/tasks/apex/batch.py

```diff
@@ -26,78 +25,105 @@ class BatchApexWait(BaseSalesforceApiTask):
     }
 
     def _run_task(self):
+        self.batches = {}  # Order matters here. Luckily Python dicts are ordered.
+
         self.poll_interval_s = int(self.options.get("poll_interval", 10))
 
         self._poll()  # will block until poll_complete
 
         self.logger.info("Job is complete.")
 
-        vals = {
-            key: value
-            for key, value in self.batch.items()
-            if key
-            in {
-                "Id",
-                "Status",
-                "ExtendedStatus",
-                "NumberOfErrors",
-                "JobItemsProcessed",
-                "TotalJobItems",
-            }
-        }
-        if not self.success:
+        summary = self.summarize_batches()
+        failed_batches = self.failed_batches()
+
+        if failed_batches:
             self.logger.info("There have been some batch failures.")
             self.logger.info("Error values:")
-            self.logger.info(repr(vals))
-            raise SalesforceException("There were import errors: %s" % repr(vals))
+            self.logger.info(repr(failed_batches))
+            raise SalesforceException(
+                f"There were import errors: {repr(failed_batches)}"
```

Comment:
Are these alway guaranteed to be import errors, or should wording be more general?

Reply:
https://github.com/SFDO-Tooling/CumulusCI/pull/1490/files#diff-defe95b2261349328fbef8dd3bceeb93R44